src/device.rs

```diff
@@ -0,0 +1,237 @@
+    pub auto_cue: Option<setting::AutoCue>,
+    /// "MASTER TEMPO" setting.
+    pub master_tempo: Option<setting::MasterTempo>,
+    /// "TEMPO RANGE" setting.
+    pub tempo_range: Option<setting::TempoRange>,
+    /// "PHASE METER" setting.
+    pub phase_meter: Option<setting::PhaseMeter>,
+
+    // MYSETTING2.DAT
+    /// "VINYL SPEED ADJUST" setting.
+    pub vinyl_speed_adjust: Option<setting::VinylSpeedAdjust>,
+    /// "JOG DISPLAY MODE" setting.
+    pub jog_display_mode: Option<setting::JogDisplayMode>,
+    /// "PAD/BUTTON BRIGHTNESS" setting.
+    pub pad_button_brightness: Option<setting::PadButtonBrightness>,
+    /// "JOG LCD BRIGHTNESS" setting.
+    pub jog_lcd_brightness: Option<setting::JogLCDBrightness>,
+    /// "WAVEFORM DIVISIONS" setting.
+    pub waveform_divisions: Option<setting::WaveformDivisions>,
+    /// "WAVEFORM / PHASE METER" setting.
+    pub waveform: Option<setting::Waveform>,
+    /// "BEAT JUMP BEAT VALUE" setting.
+    pub beat_jump_beat_value: Option<setting::BeatJumpBeatValue>,
+
+    // DJMSETTING.DAT
+    /// "CH FADER CURVE" setting.
+    pub channel_fader_curve: Option<setting::ChannelFaderCurve>,
+    /// "CROSSFADER CURVE" setting.
+    pub crossfader_curve: Option<setting::CrossfaderCurve>,
+    /// "HEADPHONES PRE EQ" setting.
+    pub headphones_pre_eq: Option<setting::HeadphonesPreEQ>,
+    /// "HEADPHONES MONO SPLIT" setting.
+    pub headphones_mono_split: Option<setting::HeadphonesMonoSplit>,
+    /// "BEAT FX QUANTIZE" setting.
+    pub beat_fx_quantize: Option<setting::BeatFXQuantize>,
+    /// "MIC LOW CUT" setting.
+    pub mic_low_cut: Option<setting::MicLowCut>,
+    /// "TALK OVER MODE" setting.
+    pub talk_over_mode: Option<setting::TalkOverMode>,
+    /// "TALK OVER LEVEL" setting.
+    pub talk_over_level: Option<setting::TalkOverLevel>,
+    /// "MIDI CH" setting.
+    pub midi_channel: Option<setting::MidiChannel>,
+    /// "MIDI BUTTON TYPE" setting.
+    pub midi_button_type: Option<setting::MidiButtonType>,
+    /// "BRIGHTNESS > DISPLAY" setting.
+    pub display_brightness: Option<setting::MixerDisplayBrightness>,
+    /// "BRIGHTNESS > INDICATOR" setting.
+    pub indicator_brightness: Option<setting::MixerIndicatorBrightness>,
+    /// "CH FADER CURVE (LONG FADER)" setting.
+    pub channel_fader_curve_long_fader: Option<setting::ChannelFaderCurveLongFader>,
+
+    // DEVSETTING.DAT
+    /// "Type of the overview Waveform" setting.
+    pub overview_waveform_type: Option<setting::OverviewWaveformType>,
+    /// "Waveform color" setting.
+    pub waveform_color: Option<setting::WaveformColor>,
+    /// "Key display format" setting.
+    pub key_display_format: Option<setting::KeyDisplayFormat>,
+    /// "Waveform Current Position" setting.
+    pub waveform_current_position: Option<setting::WaveformCurrentPosition>,
+}
+
+impl Settings {
+    fn set_mysetting(&mut self, data: &setting::MySetting) {
+        self.on_air_display = Some(data.on_air_display);
```

Comment:
fyi this would also work, though the current manual construction is also fine imo.
 ```suggestion
        self.on_air_display = data.on_air_display.into();
```